src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs

```diff
@@ -8,16 +8,19 @@ namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
     using Markdig.Syntax;
     using System;
     using System.Collections.Generic;
+    using System.Diagnostics;
     using System.Net;
     using System.Security.Cryptography;
     using System.Text;
+    using System.Text.RegularExpressions;
 
     public class ImageExtension : ITripleColonExtensionInfo
     {
         public string Name => "image";
         public bool SelfClosing => true;
-        public bool EndingTripleColons => true;
+        public bool EndingTripleColons { get; set; } = false;
         public Func<HtmlRenderer, TripleColonBlock, bool> RenderDelegate { get; private set; }
+        private List<ImageProperties> imagesList = new List<ImageProperties>();
```

Comment:
Same as above reason, this list as a member variable will have threading problems in v3, as different threads are trying to add/use the same list when building multiple document at the same time. It could be turned into a local variable inside `Render` method.

Reply:
@yufeih - This should be resolved.